# -*- coding: utf-8 -*-
"""write_pinnacle_slurm.ipynb

Automatically generated by Colaboratory.

Original file is located at
    https://colab.research.google.com/drive/1P80C_kQGmOUuzieDCGh2ePiEDzHXKtgw
"""

#! /usr/bin/bash python3
  
# This script generates a SLURM file for the AHPCC Razor cluster

# define some variables
job = 'pinnacle_04.slurm'
queue = 'onenode16core'
prefix = 'pinnacle_04.out$PBS_JOBID'
nodes = 1 
ppn = 32
wall = 3 # this is in hours


print('SBATCH -J', job)
print('SBATCH --partition comp06')
print('SBATCH -o', prefix)
print('SBATCH -e Trinity_%j.err')
print('SBATCH --mail-type=ALL')
print('SBATCH --mail-user=name@uark.edu')  
print('SBATCH --nodes=',nodes)
print('SBATCH --ntasks-per-node=', ppn)
print('SBATCH --time=',wall)
print()

print('export OMP_NUM_THREADS=32')
 
# load required modules
print('module load samtools')
print('module load jellyfish')
print('module load bowtie2')
print('module load salmon/0.8.2')
print('module load java')
print()

# cd into the directory where you're submitting this script from
print('cd $SLURM_SUBMIT_DIR')
print()

# copy files from storage to scratch
print('rsync -av RNA-R*.fastq.gz /scratch/$SLURM_JOB_ID')
print()

# cd onto the scratch disk to run the job
print('cd /scratch/$SLURM_JOB_ID/')
print()

# run the Trinity assembly
print('/share/apps/bioinformatics/trinity/trinityrnaseq-v2.11.0/Trinity --seqType fq --left RNA-R1.fastq.gz --right RNA-R2.fastq.gz --CPU 48 --max_memory 250G --trimmomatic --no_normalize_reads --full_cleanup --output trinity_Run2')
print()

# copy output files back to storage
print('rsync -av trinity_Run2 $SLURM_SUBMIT_DIR')